src/collective/mustread/interfaces.py

```diff
@@ -190,10 +192,91 @@ def who_unread(obj, force_deadline=True):
+                  did not read and their deadline (``{userid: deadline}``)
+        :rtype: dict
+        '''
+
+    def what_to_read(context=None, userid=None):
+        '''Query which content items have open read requests.
+
+        This only queries for objects who have been explicitly scheduled via
+        the ``schedule_must_read()`` method.
+        If no read requests have been scheduled an empty list is returned.
+
+        If context is given, limit the result to objects located within
+        this context. If context is ``None`` search within the plonesite.
+
+        If userid is given, limit results to objects with an open read request
+        for this userid.
+
+        @guido: if we implement the force_deadline here, we'd have an easy way
+        to list all items with missed deadlines too. alternatively we can make
+        get_report more powerful/complex
+
+        :param context: Context to search for objects with open read requests
+        :type context: Content object (must be IUUID resolvable)
+        :param userid: Userid to search for open read requests for.
+        :type userid: string
+        :returns: List of content objects with open read requests
+        :rtype: Generator
+        '''
+        # xxx guido: generator or list? (we intend to use all elements anyway)
+
+    def get_report(context=None, include_children=True, userid=None,
+                   start_date=None):
+        '''Return all mustread database entries for context.
+        (useful for exporting data or displaying detailed reports)
+
+        If context is None, defaults to plone site.
+
+        If include_children is ``True`` (which is the default), include
+        mustread records for child-objects of context.
+
+        If userid is given, only return entries for this user
+
+        If start_date is given, only return entries created after start_date
+        (``scheduled_at`` or ``read_at`` >= ``start_date``
+
+        :param context: Context to create mustread report on
+        :type context: Content object (must be IUUID resolvable)
+        :param include_children: Whether to include child-objects of context
+        :type include_children: Bool
+        :param userid: only report records for specified user
+        :type userid: string
+        :param start_date: Filter out entries older than start_date
+        :type read_at: datetime
+        :returns: List of dictionaries, each representing a mustread record
+                  ``[{userid, uid, status, read_at, deadline, scheduled_at,
+                      type, path, ...}]``
+        :rtype: Generator
+        '''
+        # xxx guido: generator or list? (we intend to use all elements anyway)
+
+    def unschedule_must_read(obj=None, userids=None):
+        '''Maintenance method to remove open must-read requests for an object
+        and/or a specified userid
+
+        @guido decide if we need/want this
```

Comment:
i can imagine that when using the mustread functionality a lot you'll end up with quite a lot of objects returned by what_to_read although their deadline has passed a long time ago (think of employees that quit, where ill and "forget" about a reminder, etc).
we'll need a view (eg in collective.contentrules.mustread) that lists open requests and also allows privileged users to undo the schedulung of an open request or mark an open request as "accepted"